ynr/apps/official_documents/static/official_documents/js/sopn_viewer.js

```diff
@@ -0,0 +1,76 @@
+function load_pages_by_range(pdf, start_page, end_page) {
```

Comment:
What's the intended browser support target for this JS.

Also, just to confirm, is this the only .js file that contains custom JS - all the other ones added in this commit are 3rd party, yeah?

Reply:
Yes, this is the only file that adds custom JS. The other files should be added with NPM really, but that's a different issue, so I'll just commit them until I move to NPM.

As for browser support…donno. I reckon >IE9. Why, what have you spotted?

Reply:
I hadn't looked at it at all. The reason I ask is because given we're not using any compatibility focussed-tooling on this project, "where do you expect this to run?" is the single most important thing to know in order to understand how to read/review this.

If you want this to run on IE9, there is a bunch of stuff that won't work here. For a start, `let` and `const` require IE11 or greater, so those would all need to be changed to `var`. Promises ( `.then()` ) aren't supported in any version of IE (even 11) - for MS users they were first introduced in Edge, so those would need to be written using callbacks to support any version of IE.

Another approach could be to use a compatibility shim/polyfill for these features. Another thing we could do is test support first e.g:

```js
if(typeof Promise !== "undefined") {
  // use pdf.js
} else {
  // drop to google docs iframe
}
```

Do we have any decent stats for what browsers logged-in users actually use? We may be worrying about a fictional use-case here.

Reply:
Looking in to it, pdf.js [requires quite a lot of things](https://github.com/mozilla/pdf.js/wiki/Required-Browser-Features) and they don't have a check internally. They do have [some tests](http://mozilla.github.io/pdf.js/features/), but implementing all of them in this code base seems silly. 

I've gone for the sledgehammer approach of just catching any exception and showing the Google Viewer. This at least provides a smoother "enhancement" for newer browsers.